src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -869,7 +869,7 @@ public string[] Segments
                     }
                     else
                     {
-                        LowLevelList<string> pathSegments = new LowLevelList<string>();
+                        var pathSegments = new ArrayBuilder<string>();
```

Comment:
LowLevelList had a different allocation pattern, due to using arrays of length 0, 4, 8, 16, etc.  Do we know if/how this change affects the performance here for various common use cases?
cc: @davidsh, @cipop